training/training.pages.inc

```diff
@@ -5,16 +5,151 @@
  */
 
 /**
- * Renders example form.
+ * Renders a page with listing of text message.
  *
- * @param array $arg
- *  Id of example form.
+ * Selects rows from table 'text_message' Renders tables with pager and sorting.
  *
- * @return array|mixed
- *   The form array.
+ * @return array
+ *   Renderable array.
  */
-function training_example_form_callback($arg) {
-  return drupal_get_form($arg, array());
+function training_text_message_callback() {
+  $header = array(
+    array('data' => t('Num'), 'field' => 'id'),
+    array('data' => t('Author'), 'field' => 'name'),
+    array('data' => t('Title'), 'field' => 'title'),
+    array('data' => t('Text'), 'field' => 'body'),
+    array('data' => t('Edit')),
+  );
+  $rows = array();
+  $query = db_select('text_message', 'tm');
+  $query->innerJoin('users', 'u', 'tm.uid = u.uid');
+  $query
+    ->fields('tm', array('id', 'title', 'body'))
+    ->fields('u', array('name'));
+  $result = $query
```

Comment:
Sorry, but pager works only this way:
$query
    ->fields('tm', array('id', 'title', 'body'))
    ->fields('u', array('name'));
  $result = $query
    ->extend('PagerDefault')
    ->limit(5)
    ->extend('TableSort')
    ->orderByHeader($header)
    ->execute()->fetchAll();
i dont know why(